

<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

  <meta name="viewport" content="width=device-width,initial-scale=1">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <meta name="lang:clipboard.copy" content="Copy to clipboard">
  <meta name="lang:clipboard.copied" content="Copied to clipboard">
  <meta name="lang:search.language" content="en">
  <meta name="lang:search.pipeline.stopwords" content="True">
  <meta name="lang:search.pipeline.trimmer" content="True">
  <meta name="lang:search.result.none" content="No matching documents">
  <meta name="lang:search.result.one" content="1 matching document">
  <meta name="lang:search.result.other" content="# matching documents">
  <meta name="lang:search.tokenizer" content="[\s\-]+">

  
    <link href="https://fonts.gstatic.com/" rel="preconnect" crossorigin>
    <link href="https://fonts.googleapis.com/css?family=Roboto+Mono:400,500,700|Roboto:300,400,400i,700&display=fallback" rel="stylesheet">

    <style>
      body,
      input {
        font-family: "Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif
      }

      code,
      kbd,
      pre {
        font-family: "Roboto Mono", "Courier New", Courier, monospace
      }
    </style>
  

  <link rel="stylesheet" href="../_static/stylesheets/application.css"/>
  <link rel="stylesheet" href="../_static/stylesheets/application-palette.css"/>
  <link rel="stylesheet" href="../_static/stylesheets/application-fixes.css"/>
  
  <link rel="stylesheet" href="../_static/fonts/material-icons.css"/>
  
  <meta name="theme-color" content="#3f51b5">
  <script src="../_static/javascripts/modernizr.js"></script>
  
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXX"></script>
<script>
    window.dataLayer = window.dataLayer || [];

    function gtag() {
        dataLayer.push(arguments);
    }

    gtag('js', new Date());

    gtag('config', 'UA-XXXXX');
</script>
  
  
    <title>Alignment &#8212; CPT3 1.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/material.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="PET divide" href="PETdivide.html" />
    <link rel="prev" title="Linker filter" href="Linkerfilter.html" />



  
   

  </head>
  <body dir=ltr
        data-md-color-primary=blue data-md-color-accent=light-blue>
  
  <svg class="md-svg">
    <defs data-children-count="0">
      
      <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
      
    </defs>
  </svg>
  
  <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer">
  <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search">
  <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
  <a href="#function/Alignment" tabindex="1" class="md-skip"> Skip to content </a>
  <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex navheader">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../index.html" title="CPT3 1.1 documentation"
           class="md-header-nav__button md-logo">
          
            &nbsp;
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          <span class="md-header-nav__topic">CPT3 Usage</span>
          <span class="md-header-nav__topic"> Alignment </span>
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
        
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" action="../search.html" method="get" name="search">
      <input type="text" class="md-search__input" name="q" placeholder="Search"
             autocapitalize="off" autocomplete="off" spellcheck="false"
             data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>

      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            <a href="https://github.com/GuoliangLi-HZAU/ChIA-PET_Tool_V3" title="Go to repository" class="md-source" data-md-source="github">

    <div class="md-source__icon">
      <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 24 24" width="28" height="28">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    CPT3 resource
  </div>
</a>
          </div>
        </div>
      
      
  
  <script src="../_static/javascripts/version_dropdown.js"></script>
  <script>
    var json_loc = "../"versions.json"",
        target_loc = "../../",
        text = "Versions";
    $( document ).ready( add_version_dropdown(json_loc, target_loc, text));
  </script>
  

    </div>
  </nav>
</header>

  
  <div class="md-container">
    
    
    
  <nav class="md-tabs" data-md-component="tabs">
    <div class="md-tabs__inner md-grid">
      <ul class="md-tabs__list">
          <li class="md-tabs__item"><a href="../index.html" class="md-tabs__link">CPT3 1.1 documentation</a></li>
      </ul>
    </div>
  </nav>
    <main class="md-main">
      <div class="md-main__inner md-grid" data-md-component="container">
        
          <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
            <div class="md-sidebar__scrollwrap">
              <div class="md-sidebar__inner">
                <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../index.html" title="CPT3 1.1 documentation" class="md-nav__button md-logo">
      
        <img src="../_static/" alt=" logo" width="48" height="48">
      
    </a>
    <a href="../index.html"
       title="CPT3 1.1 documentation">CPT3 Usage</a>
  </label>
    <div class="md-nav__source">
      <a href="https://github.com/GuoliangLi-HZAU/ChIA-PET_Tool_V3" title="Go to repository" class="md-source" data-md-source="github">

    <div class="md-source__icon">
      <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 24 24" width="28" height="28">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    CPT3 resource
  </div>
</a>
    </div>
  
  

  
  <ul class="md-nav__list">
    <li class="md-nav__item">
    
      <span class="md-nav__link caption"><span class="caption-text">Basic Use</span></span>
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="install.html" class="md-nav__link">Installation</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="example.html" class="md-nav__link">Example Data</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="bdindex.html" class="md-nav__link">Build index</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Pipeline.html" class="md-nav__link">Pipeline</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="Linkerfilter.html" class="md-nav__link">Linker filter</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    <label class="md-nav__link md-nav__link--active" for="__toc"> Alignment </label>
    
      <a href="#" class="md-nav__link md-nav__link--active">Alignment</a>
      
        
<nav class="md-nav md-nav--secondary">
    <label class="md-nav__title" for="__toc">Contents</label>
  <ul class="md-nav__list" data-md-scrollfix="">
        <li class="md-nav__item"><a href="#function-alignment--page-root" class="md-nav__link">Alignment</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#cpt3-align" class="md-nav__link">CPT3  align</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#long-read-or-short-read" class="md-nav__link">long-read or short-read</a>
        </li>
        <li class="md-nav__item"><a href="#parameters" class="md-nav__link">Parameters</a>
        </li></ul>
            </nav>
        </li></ul>
            </nav>
        </li>
    
<li class="md-nav__item"><a class="md-nav__extra_link" href="../_sources/function/Alignment.rst.txt">Show Source</a> </li>

  </ul>
</nav>
      <ul class="md-nav__list"> 
    <li class="md-nav__item">
    
    
      <a href="#cpt3-align" class="md-nav__link">CPT3  align</a>
      <ul class="md-nav__list"> 
    <li class="md-nav__item">
    
    
      <a href="#long-read-or-short-read" class="md-nav__link">long-read or short-read</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="#parameters" class="md-nav__link">Parameters</a>
      
    
    </li></ul>
    
    </li></ul>
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="PETdivide.html" class="md-nav__link">PET divide</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="peakcall.html" class="md-nav__link">Peak calling</a>
      
    
    </li>
    <li class="md-nav__item">
    
    
      <a href="loopcall.html" class="md-nav__link">Loop calling</a>
      
    
    </li>
  </ul>
  

</nav>
              </div>
            </div>
          </div>
          <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
            <div class="md-sidebar__scrollwrap">
              <div class="md-sidebar__inner">
                
<nav class="md-nav md-nav--secondary">
    <label class="md-nav__title" for="__toc">Contents</label>
  <ul class="md-nav__list" data-md-scrollfix="">
        <li class="md-nav__item"><a href="#function-alignment--page-root" class="md-nav__link">Alignment</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#cpt3-align" class="md-nav__link">CPT3  align</a><nav class="md-nav">
              <ul class="md-nav__list">
        <li class="md-nav__item"><a href="#long-read-or-short-read" class="md-nav__link">long-read or short-read</a>
        </li>
        <li class="md-nav__item"><a href="#parameters" class="md-nav__link">Parameters</a>
        </li></ul>
            </nav>
        </li></ul>
            </nav>
        </li>
    
<li class="md-nav__item"><a class="md-nav__extra_link" href="../_sources/function/Alignment.rst.txt">Show Source</a> </li>

<li id="searchbox" class="md-nav__item"></li>

  </ul>
</nav>
              </div>
            </div>
          </div>
        
        <div class="md-content">
          <article class="md-content__inner md-typeset" role="main">
            
  <section id="alignment">
<h1 id="function-alignment--page-root">Alignment<a class="headerlink" href="#function-alignment--page-root" title="Permalink to this heading">¶</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#cpt3-align" id="id1">CPT3  align</a></p>
<ul>
<li><p><a class="reference internal" href="#long-read-or-short-read" id="id2">long-read or short-read</a></p></li>
<li><p><a class="reference internal" href="#parameters" id="id3">Parameters</a></p></li>
</ul>
</li>
</ul>
</nav>
<section id="cpt3-align">
<h2 id="cpt3-align"><a class="toc-backref" href="#id1" role="doc-backlink">CPT3  align</a><a class="headerlink" href="#cpt3-align" title="Permalink to this heading">¶</a></h2>
<section id="long-read-or-short-read">
<h3 id="long-read-or-short-read"><a class="toc-backref" href="#id2" role="doc-backlink">long-read or short-read</a><a class="headerlink" href="#long-read-or-short-read" title="Permalink to this heading">¶</a></h3>
<p>ChIA-PET, HiCHIP sequencing paired-end data alignment:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
    java -jar ChIA-PET.jar --fastq1 test_long_1.fastq.gz --fastq2 test_long_2.fastq.gz <span class="se">\</span>
    --minimum_linker_alignment_score <span class="m">14</span> --GENOME_INDEX hg38.fa --GENOME_LENGTH 3E9 <span class="se">\</span>
    --CHROM_SIZE_INFO ChIA-PET_Tool_V3/chromInfo/hg38.chromSize.txt <span class="se">\</span>
    --CYTOBAND_DATA ChIA-PET_Tool_V3/chromInfo/hg19_cytoBandIdeo.txt --SPECIES <span class="m">1</span> <span class="se">\</span>
    --output test_long --prefix GM12878 --thread <span class="m">8</span> --fastp fastp <span class="se">\</span>
    --start_step <span class="m">1</span> --stop_step <span class="m">2</span>
</pre></div>
</div>
</section>
<section id="parameters">
<h3 id="parameters"><a class="toc-backref" href="#id3" role="doc-backlink">Parameters</a><a class="headerlink" href="#parameters" title="Permalink to this heading">¶</a></h3>
<table>
<thead>
<tr class="row-odd"><th class="head" colspan="9"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="9"><p>System.out.println("Usage: java -jar &lt;path of ChIA_PET.jar&gt; [options]</p></td>
</tr>
<tr class="row-odd"><td colspan="9"><p>Necessary options:</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--fastq1</p></td>
<td colspan="7"><p>path of read1 fastq file</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--fastq2</p></td>
<td colspan="7"><p>path of read2 fastq file</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--autolinker</p></td>
<td colspan="7"><p>detect linker by our program, then no need provide --linker and --mode
paramater.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--mode</p></td>
<td colspan="7"><p>mode of tool, 0: short read; 1: long read, need for ChIA-PET data</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--linker</p></td>
<td colspan="7"><p>path of linker file, need for ChIA-PET mode</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--fastp</p></td>
<td colspan="7"><p>fastp path, strong suggest for ChIA-PET data.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--skipheader</p></td>
<td colspan="7"><p>skip header N reads for detect linker, default 1000000.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--linkerreads</p></td>
<td colspan="7"><p>N reads used for detect linker, default 100000.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--hichip</p></td>
<td colspan="7"><p>Y(es) or N(o)[default] or O(nly print restriction site file without run
other step), need for hichip data</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--ligation_site</p></td>
<td colspan="7"><p>It can be the name of restriction enzyme, such as HindIII, MboI, DpnII,
Bglii, Sau3AI, Hinf1, NlaIII, AluI or the site of enzyme digestion,
A^AGCTT, ^GATC, ^GATC, A^GATCT, G^ANTC, CATG^, AG^CT or others.
multipe restriction enzyme can be seperated by comma, such as
G^ANTC,^GATC. restriction site with '^' and contains 'ATCG' without other
character!!! if the genomic enzyme digestion file --restrictionsiteFile
is provided, this parameter does not need to be provided. only needed for
hichip data</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--ResRomove</p></td>
<td colspan="7"><p>Y or N, whether remove PET in same restriction contig. default: Y</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>--restrictionsiteFile</p></td>
<td colspan="6"><p>restriction site file, can be genarated while has --ligation_site and
without this paramater or provide restriction enzyme information with
--ligation_site, we will automatically generate the file. only needed
for hichip data</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--genomefile</p></td>
<td colspan="7"><p>genome fasta file path, needed for with --ligation_site and without
--restrictionsiteFile only needed for hichip data</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>--minfragsizeMinimum</p></td>
<td colspan="6"><p>restriction fragment length to consider, default 20</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--maxfragsize</p></td>
<td colspan="7"><p>Maximum restriction fragment length to consider, default 1000000</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--minInsertsize</p></td>
<td colspan="7"><p>Minimum restriction fragment skip of mapped reads to consider, default 1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--fqmode</p></td>
<td colspan="7"><p>single-end or paired-end (default), only required --fastq1 when
single-end mode for ChIA-PET data</p></td>
</tr>
<tr class="row-odd"><td colspan="7"><p>--minimum_linker_alignment_score</p></td>
<td colspan="2"><p>minimum alignment score, 14 default</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--GENOME_INDEX</p></td>
<td colspan="7"><p>the path of BWA index</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--GENOME_LENGTH</p></td>
<td colspan="7"><p>the number of base pairs in the whole genome</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--CHROM_SIZE_INFO</p></td>
<td colspan="7"><p>the file that contains the length of each chromosome, example file is in
ChIA-PET_Tool_V3/chrInfo, this is necessary for &gt; step 2 analysis.
Note. please make sure chromosome name in this file is same as name in
genome file!!!</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--CYTOBAND_DATA</p></td>
<td colspan="7"><p>the ideogram data used to plot intra-chromosomal peaks and interactions,
example file is in ChIA-PET_Tool_V3/chrInfo</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--SPECIES</p></td>
<td colspan="7"><p>1: human; 2: mouse; 3: others</p></td>
</tr>
<tr class="row-odd"><td colspan="9"><p>Other options:</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--start_step</p></td>
<td colspan="7"><p>start with which step, 1: linker filtering; 2: mapping to genome;
3: removing redundancy; 4: categorization of PETs; 5: peak calling;
6: interaction calling; 7: visualizing, default: 1</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--stop_step</p></td>
<td colspan="7"><p>stop with which step, 1: linker filtering; 2: mapping to genome;
3: removing redundancy; 4: categorization of PETs; 5: peak calling;
6: interaction calling; 7: visualizing, default: 100, should be bigger
than --start_step</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--output</p></td>
<td colspan="7"><p>path of output, default: ChIA-PET_Tool_V3/output</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--prefix</p></td>
<td colspan="7"><p>prefix of output files, default: out</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--minimum_tag_length</p></td>
<td colspan="7"><p>minimum tag length, default: 18</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--maximum_tag_length</p></td>
<td colspan="7"><p>maximum tag length, default: 1000</p></td>
</tr>
<tr class="row-even"><td colspan="5"><p>--minSecondBestScoreDiff</p></td>
<td colspan="4"><p>the score difference between the best-aligned and the second-best
aligned linkers, default: 3</p></td>
</tr>
<tr class="row-odd"><td colspan="8"><p>--output_data_with_ambiguous_linker_info</p></td>
<td><p>whether to print the linker-ambiguity PETs,
0: not print; 1: print, default: 1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--printreadID</p></td>
<td colspan="7"><p>write read ID to bedpe file, default: N</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--printallreads</p></td>
<td colspan="7"><p>print all reads no matter strand, default: 0[print all]; 1, only print
valid strand reads.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--search_all_linker</p></td>
<td colspan="7"><p>search all linkers in reads or just search one time, default: N</p></td>
</tr>
<tr class="row-odd"><td><p>--thread</p></td>
<td colspan="8"><p>the number of threads used in linker filtering and mapping to genome, default: 1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--MAPPING_CUTOFF</p></td>
<td colspan="7"><p>cutoff of mapping quality score for filtering out low-quality or
multiply-mapped reads, default: 20</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--MERGE_DISTANCE</p></td>
<td colspan="7"><p>the distance limit to merge the PETs with similar mapping locations,
default: 2</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>--SELF_LIGATION_CUFOFF</p></td>
<td colspan="6"><blockquote>
<div><p>the distance threshold between self-ligation PETs and intra-chromosomal</p>
</div></blockquote>
<p>inter-ligation PETs, default: 8000 for ChIA, and 1000 for HiChIP</p>
</td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--EXTENSION_LENGTH</p></td>
<td colspan="7"><p>the extension length from the location of each tag, default: 500,
1500 suggest for single-end mode</p></td>
</tr>
<tr class="row-even"><td colspan="4"><p>--MIN_COVERAGE_FOR_PEAK</p></td>
<td colspan="5"><p>the minimum coverage to define peak regions, default: 5</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--PEAK_MODE</p></td>
<td colspan="7"><p>1: peak region mode, which takes all the overlapping PET regions above
the coverage threshold as peak regions; 2: peak summit mode, which takes
the highest coverage of overlapping regions as peak regions, default: 2</p></td>
</tr>
<tr class="row-even"><td colspan="6"><p>--MIN_DISTANCE_BETWEEN_PEAK</p></td>
<td colspan="3"><p>the minimum distance between two peaks, default: 500</p></td>
</tr>
<tr class="row-odd"><td colspan="6"><p>--GENOME_COVERAGE_RATIO</p></td>
<td colspan="3"><p>the estimated proportion of the genome covered by the reads,
default: 0.8</p></td>
</tr>
<tr class="row-even"><td colspan="6"><p>--PVALUE_CUTOFF_PEAK</p></td>
<td colspan="3"><p>p-value to filter peaks that are not statistically significant,
default: 0.00001</p></td>
</tr>
<tr class="row-odd"><td colspan="6"><p>--INPUT_ANCHOR_FILE</p></td>
<td colspan="3"><p>a file which contains user-specified anchors for interaction
calling, default: null</p></td>
</tr>
<tr class="row-even"><td colspan="9"><p>--PVALUE_CUTOFF_INTERACTION | p-value to filter false positive interactions, default: 0.5</p></td>
</tr>
<tr class="row-odd"><td><p>--zipbedpe</p></td>
<td colspan="8"><p>gzip bedpe related file, after analysis done. default: N. Y for gzip, N for not.</p></td>
</tr>
<tr class="row-even"><td><p>--zipsam</p></td>
<td colspan="8"><p>Convert sam file to bam, after analysis done. default: N</p></td>
</tr>
<tr class="row-odd"><td><p>--deletesam</p></td>
<td colspan="8"><p>Delete sam files. default: N</p></td>
</tr>
<tr class="row-even"><td><p>--keeptemp</p></td>
<td colspan="8"><p>Keep temp sam and bedpe file. default: N</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--map_ambiguous</p></td>
<td colspan="7"><p>Also mapping ambiguous reads without linker. default: N</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--skipmap</p></td>
<td colspan="7"><p>Skip mapping read1 and read2, start from paired R1.sam and R2.sam, only
valid in HiChIP mode now. default: N</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--macs2</p></td>
<td colspan="7"><p>macs2 path, using macs2 callpeak to detect anchor peak with alignment
file. default: N</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--nomodel</p></td>
<td colspan="7"><p>macs2 parameter, Whether or not to build the shifting model in macs2.
default: N</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--shortestP</p></td>
<td colspan="7"><p>extend and keep shorest peak length longer than N for loop calling,
suggest 1500, user can set 0 to skip this step. default: 1500</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--shortestA</p></td>
<td colspan="7"><p>extend and keep shorest anchor length longer than N for loop calling,
user can set 0 to skip this step. default: 0</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>--XOR_cluster</p></td>
<td colspan="7"><p>Whether keep loops if only one side of anchor is overlap with peak.
default: N</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>--addcluster</p></td>
<td colspan="7"><p>Keep all regions with more than 2 count reads as potential anchor for
calling loop. default: N. if peaks number of macs2 smaller than 10000,
this paramater will work automaticly.</p></td>
</tr>
</tbody>
</table>
<p>Note: To use CPT3, you need to first index the genome with <a class="reference internal" href="bdindex.html"><span class="doc">Build index</span></a>.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="https://github.com/GuoliangLi-HZAU/ChIA-PET_Tool_V3">on github</a>.</p>
</div>
</section>
</section>
</section>


          </article>
        </div>
      </div>
    </main>
  </div>
  <footer class="md-footer">
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
          
            <a href="Linkerfilter.html" title="Linker filter"
               class="md-flex md-footer-nav__link md-footer-nav__link--prev"
               rel="prev">
              <div class="md-flex__cell md-flex__cell--shrink">
                <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
              </div>
              <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
                <span class="md-flex__ellipsis">
                  <span
                      class="md-footer-nav__direction"> Previous </span> Linker filter </span>
              </div>
            </a>
          
          
            <a href="PETdivide.html" title="PET divide"
               class="md-flex md-footer-nav__link md-footer-nav__link--next"
               rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span
                class="md-flex__ellipsis"> <span
                class="md-footer-nav__direction"> Next </span> PET divide </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink"><i
                class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          
        </a>
        
      </nav>
    </div>
    <div class="md-footer-meta md-typeset">
      <div class="md-footer-meta__inner md-grid">
        <div class="md-footer-copyright">
          <div class="md-footer-copyright__highlight">
              &#169; Copyright 2022, momocoding.
              
          </div>
            Created using
            <a href="http://www.sphinx-doc.org/">Sphinx</a> 5.2.0.post0.
             and
            <a href="https://github.com/bashtage/sphinx-material/">Material for
              Sphinx</a>
        </div>
      </div>
    </div>
  </footer>
  <script src="../_static/javascripts/application.js"></script>
  <script>app.initialize({version: "1.0.4", url: {base: ".."}})</script>
  </body>
</html>